pkg/kubeapiserver/admission/initializer.go

```diff
@@ -67,6 +75,14 @@ func (i pluginInitializer) Initialize(plugin admission.Interface) {
 		wants.SetInternalClientSet(i.internalClient)
 	}
 
+	if wants, ok := plugin.(WantsToRun); ok {
```

Comment:
If I write an admission controller that implements `WantsToRun` would I expect the `Run` to be the last operation invoked?

If I need access to the `InformerFactory` or any other information we provide in a `Wants*` extension in order to start my embedded controller, then I would be out of luck.

I think we need to do either of the following:

1. make clear that this is the last `Wants*` interface invoked
1. change the consumers of this interface to not start embedded controller loops in `Run` and instead, update those controllers to use a `WaitForReady` function similar to https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/admission/namespace/lifecycle/admission.go#L107 where the wait check would ensure the controller is started.

I am curious what @deads2k thinks here.  I am biased to not guaranteeing ordering of invocation, but it does potentially make authoring admission plug-ins trickier.